DllImportGenerator/DllImportGenerator/Marshalling/CustomNativeTypeMarshaler.cs

```diff
@@ -262,7 +253,7 @@ public bool UsesNativeIdentifier(TypePositionInfo info, StubCodeContext context)
             }
             if (context.PinningSupported)
             {
-                if (!info.IsByRef && (_marshallingMethods & SupportedMarshallingMethods.Pinning) == 0)
+                if (!info.IsByRef && (_marshallingMethods & SupportedMarshallingMethods.Pinning) != 0)
```

Comment:
I assume a test should have failed here. If not, we should consider why not and if a class of bugs are missed by our testing.

Reply:
Yes, when I was doing local testing after consuming Elinor's change that makes automatically declaring `nativeIdentifier` conditional on `UsesNativeIdentifier` returning true, I saw multiple unit test failures as well as failures building the integration tests due to this mistake.